strongbox-web-core/src/test/java/org/carlspring/strongbox/controllers/layout/npm/NpmArtifactControllerTest.java

```diff
@@ -125,4 +127,55 @@ public void testPackageCommonFlow(@NpmRepository(repositoryId = REPOSITORY_RELEA
                .header(HttpHeaders.CONTENT_LENGTH, equalTo(String.valueOf(Files.size(packagePath))));
     }
 
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
+    @Test
+    public void addUserTest(@NpmRepository(repositoryId = REPOSITORY_RELEASES)
+                                    Repository repository)
+    {
+        String url = getContextBaseUrl() + "/storages/{storageId}/{repositoryId}/" + NpmLayoutProvider.NPM_USER_PATH;
+
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
+
+        NpmUser strongboxUser1 = new NpmUser("admin", "password");
+        NpmUser strongboxUser2 = new NpmUser("deployer", "password");
+        NpmUser nonStrongboxUser = new NpmUser("notARealUser", "notARealPassword");
+
+        //can login with strongbox user
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(strongboxUser1)
+               .when()
+               .put(url, storageId, repositoryId, strongboxUser1.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.CREATED.value());
+
+        //can login with another strongbox user after login
+        mockMvc.contentType(MediaType.APPLICATION_JSON_VALUE)
+               .body(strongboxUser2)
+               .when()
+               .put(url, storageId, repositoryId, strongboxUser2.getName())
+               .peek()
+               .then()
+               .statusCode(HttpStatus.CREATED.value());
+
+        //can login when the url username differs from the body
```

Comment:
we shouldn't be able to login in this case.
also we shouldn't be able to login with `Authorization: Basic` header if credentials not the same as in URL and Body